pkg/packetfilter/packetfilter.go

```diff
@@ -138,6 +136,37 @@ type ChainIPHook struct {
 	JumpRule *Rule
 }
 
+type SetFamily uint32
+
+const (
+	SetFamilyV4 SetFamily = iota
+	SetFamilyV6
+)
+
+// named set
+type NamedSet struct {
+	// Name is the set name.
+	Name string
+	// SetType specifies the named type.
+	SetType string
+	// nftables named set attached to tables
+	Table TableType
+	// SetFamily specifies the protocol family of the IP addresses to be stored in the set.
+	// The default is IPv4.
+	Family SetFamily
+}
+
+type Named interface {
```

Comment:
I think this should be called `NamedSet` as this is what consumers will store and use continuously.  The struct above is referenced once as input so I suggest `SetConfig` or `SetInfo`.

```suggestion
type NamedSet interface {
```